common/src/ws_rpc_client.rs

```diff
@@ -0,0 +1,251 @@
+use log::info;
+use serde_json::Value;
+use sp_core::H256 as Hash;
+use std::{
+    sync::{
+        mpsc::{channel, Sender as ThreadOut},
+        Arc, Mutex,
+    },
+    thread,
+    thread::JoinHandle,
+};
+use substrate_api_client::{
+    rpc::{
+        json_req,
+        ws_client::{
+            on_extrinsic_msg_submit_only, on_extrinsic_msg_until_broadcast,
+            on_extrinsic_msg_until_finalized, on_extrinsic_msg_until_in_block,
+            on_extrinsic_msg_until_ready, on_get_request_msg, OnMessageFn, RpcClient,
+        },
+    },
+    ApiClientError, ApiResult, FromHexString, RpcClient as RpcClientTrait, XtStatus,
+};
+use ws::{connect, Handler, Message, Result as WsResult, Sender as WsSender};
+
+pub struct WsRpcClient {
+    mux: Mutex<()>,
+    next_handler: Arc<Mutex<Option<RpcClient>>>,
```

Comment:
would `Option<Arc<Mutex<RpcClient>>>` work?

Reply:
I am sharing this `Option` between threads (`Handler for WsHandler`) so I think I need this `Mutex` around it.